### PR TITLE
Cancel an edit

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-62

It should be possible for users to cancel an in-progress edit.

This should destroy the EditRegistration without saving any of the modified data to the original Registration.

### CODE REVIEW

**Comment by irisfaraway** [comment]
CodeClimate has a gripe about the number of methods in the EditFormsController. Unfortunately this is down to the number of different state machines transitions possible from that state. I'm not sure how we can modify this but would appreciate ideas for refactoring (if anyone thinks it is worth it).

File: app/views/waste_exemptions_engine/edit_cancelled_forms/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (cintamani) [inline app/views/waste_exemptions_engine/edit_cancelled_forms/new.html.erb]: [NIT]
```suggestion
```

File: app/views/waste_exemptions_engine/edit_cancelled_forms/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <h1 class="heading-large"><%= t(".error_heading") %></h1>
+
+  <% @edit_cancelled_form.errors.full_messages.each do |message| %>
+    <li><%= message %></li>
+  <% end %>
+
+<% else %>
+
+  <%= form_for(@edit_cancelled_form) do |f| %>
+    <%= render("waste_exemptions_engine/shared/errors", object: @edit_cancelled_form) %>
```
Comment (cintamani) [inline app/views/waste_exemptions_engine/edit_cancelled_forms/new.html.erb]:
```suggestion
```

Comment (irisfaraway) [reply app/views/waste_exemptions_engine/edit_cancelled_forms/new.html.erb]: This had the same issues as the EditCompleteForm from the other PR. So I've done a matching fix: https://github.com/DEFRA/waste-exemptions-engine/pull/142/commits/4c00561259db806438981d2cec23a3b222b90268

**Comment by irisfaraway** [comment]
@cintamani Agree it was a pretty awkward name – I tried a few things but was getting hung up on starting it with "Edit". Have gone with "ConfirmEditCancelledForm" instead.

**Comment by irisfaraway** [comment]
Actually, hold off on reviewing this for now – need to check the styling.

**Comment by irisfaraway** [comment]
Styling added (although it won't be visible until https://github.com/DEFRA/waste-exemptions-front-office/pull/102 and https://github.com/DEFRA/waste-exemptions-back-office-ta/pull/152 are merged) – ready for a review!

**Pull request merged**